browser/src/layer/tile/CanvasTileUtils.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+declare var L: any;
+
+namespace cool {
+	export abstract class CanvasTileUtils {
+		public static unpremultiply(
+			data: Uint8Array,
+			byteLength: number,
+			byteOffset: number = 0,
+		): void {
+			for (var i = byteOffset; i < byteOffset + byteLength; i += 4) {
+				// premultiplied rgba -> unpremultiplied rgba
+				var alpha = data[i + 3];
+				if (alpha === 0) {
+					data.fill(0, i, i + 3);
+				} else if (alpha !== 255) {
+					data[i] = Math.ceil((data[i] * 255) / alpha);
```

Comment:
We should probably use a 64k byte lookup table here as some of the core LibreOffice code does =)